support/oidc-discovery-provider/config.go

```diff
@@ -23,10 +23,17 @@ type Config struct {
 	// LogRequests is a debug option that logs all incoming requests
 	LogRequests bool `hcl:"log_requests"`
 
-	// Domain is the domain this provider will be hosted under. It is used
-	// as the domain when building the JWKS URI. It is also used when obtaining
-	// obtaining certs via ACME (unless InsecureAddr is specified).
-	Domain string
```

Comment:
We need to keep the old configurable or we will break existing deployments on upgrade. I think the right path forward is add a warning log if the `domain` configurable is used saying something like:

```
The domain configurable is deprecated and will be removed in a future release. Please use the domains configurable instead.
```

We should fail if both `domains` and `domain` are set.

If `domain` is set, I think we should avoid the new domain validation that was added to the handler. This gives us a pretty good migration story towards strong domain validation without breaking existing deployments.

Reply:
Done…

If domain is set we skip the domain validation in the handler.

If domain setting is configured we will copy the value in the Domains setting so all logic can be based on the new setting, to reduce amount of refactors when moving the deprecated setting out.